### DISCUSSION TITLE
Tracing back deprecations

### DISCUSSION THREAD

**Discussion by tjveldhuizen** [symfony/symfony#45052] [Q&A]
We're working on a quite large project, which has a lot of legacy code not covered by tests. Unfortunately it's not to be expected to create tests for all code in the short term, but I prefer to keep up with the Symfony versions. Therefore, I've configured a log file which catches the deprecations. Now and then, I have a look at this log and try to change our code to get rid of the deprecation messages.

At the moment I'm trying to get rid of ```Since symfony/http-foundation 5.1: Retrieving a non-string value from "Symfony\Component\HttpFoundation\InputBag::get()" is deprecated, and will throw a "Symfony\Component\HttpFoundation\Exception\BadRequestException" exception in Symfony 6.0, use "Symfony\Component\HttpFoundation\InputBag::all($key)" instead```, however I have not been able to find which `get()` triggers it.

I wonder if someone can suggest a good work flow, to relate the entry in the deprecation log to (at least) the URL that triggers the request. As soon as I can reproduce it, it should not be hard to find the exact location of the issue.

**Comment by ruudk** [comment]
+1 one this.

Even when running 1 unit test that uses some code from other packages, it's often very hard to pinpoint the exact location of the usage of a function or method. Having the file : line number is crucial.

**Reply by tjveldhuizen** [reply]
File and line number would be perfect, but when I'm able to reproduce the request it's not too difficult to find the location that triggered the deprecation using xdebug (by setting a breakpoint at the deprecation).

**Reply by ruudk** [reply]
Sure, with a debugger you can find most things, but having to find 10-50 of these different deprecations each new PHP version, is still a lot of work.

**Comment by tjveldhuizen** [comment]
Because this thread is getting lower on the page already, I've been looking who is/are responding to the most recent PHPUnit Bridge related issues and PRs. I hope @derrabus or @nicolas-grekas is able and willing to shed some light on this question.

**Comment by nicolas-grekas** [comment]
I guess this tells how?
https://symfony.com/doc/current/components/phpunit_bridge.html#display-the-full-stack-trace

**Reply by tjveldhuizen** [reply]
Thanks for your reply. It does give sufficient details when using PHPUnit, but in my case I [log the deprecation in my production environment ](https://stackoverflow.com/a/48827892) due to the lack of all-covering tests. It's not watertight, but better then nothing. I wonder if there is a way to log more details, so I'm able to trace the location where the deprecated code is triggered.

By the way: now I'm writing this reply, I realize that logging the deprecations this way is probably not related to PHPUnit or the bridge...
